### PR TITLE
Add SLAS tenant APIs

### PR DESCRIPTION
**Pull request by hnestmann**

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -1926,6 +1926,60 @@ program
         console.log();
     });
 
+program
```
Comment (tobiaslohr) [inline cli.js:1929]: Do we want to allow someone passing the full JSON with the tenant info in there as a short hand version in alternative to passing individual properties through parameters? E.g. 

```
{
  "channels": [
    "SiteGenesis",
    "OtherSite"
  ],
  "clientId": "client-id1",
  "ecomSite": "SiteGenesis",
  "ecomTenant": "test_prd",
  "name": "ShopperClient1",
  "privateClient": true,
  "redirectUri": "http://localhost:9010/callback",
  "scopes": [
    "sfcc.products",
    "sfcc.catalogs",
    "sfcc.products.ro"
  ],
  "secret": "secret12345"
}
```

File: cli.js [hunk 2]
```diff
@@ -1926,6 +1926,60 @@ program
...
+        await slas.cli.tenant.add(options.tenant, options.shortcode, options.tenantdescription, options.merchantname, options.contact, options.email);
+
+    }).on('--help', function() {
+        console.log();
+    });
+
+program
+    .command('slas:tenant:get')
+    .description('Gets a SLAS tenant from given organisation')
+    .requiredOption('--tenant <tenant>', 'the tenant id to add')
```
Comment (tobiaslohr) [inline cli.js:1953]: Minor description change needed

File: cli.js [hunk 3]
```diff
@@ -1926,6 +1926,60 @@ program
...
+        await slas.cli.tenant.get(options.tenant, options.shortcode);
+
+    }).on('--help', function() {
+        console.log();
+    });
+
+program
+    .command('slas:tenant:delete')
+    .description('deletes a SLAS tenant from given organisation')
+    .requiredOption('--tenant <tenant>', 'the tenant id to add')
```
Comment (tobiaslohr) [inline cli.js:1970]: Minor description change needed

File: cli.js [hunk 4]
```diff
@@ -1926,6 +1926,60 @@ program
...
+    .requiredOption('--shortcode <shortcode>', 'the organizations short code')
+    .option('-j, --json', 'Formats the output in json')
+    .action(async function(options) {
+
+        var asJson = ( options.json ? options.json : false );
```
Comment (tobiaslohr) [inline cli.js:1975]: Do you use the `asJson` as all? If not we should remove it and from the options

File: cli.js [hunk 5]
```diff
@@ -1926,6 +1926,60 @@ program
...
+
+program
+    .command('slas:tenant:get')
+    .description('Gets a SLAS tenant from given organisation')
+    .requiredOption('--tenant <tenant>', 'the tenant id to add')
+    .requiredOption('--shortcode <shortcode>', 'the organizations short code')
+    .option('-j, --json', 'Formats the output in json')
+    .action(async function(options) {
+
+        var asJson = ( options.json ? options.json : false );
```
Comment (tobiaslohr) [inline cli.js:1958]: Do you use the `asJson` as all? If not we should remove it and from the options

File: cli.js [hunk 6]
```diff
@@ -1926,6 +1926,60 @@ program
...
+    .option('--tenant <tenant>', 'the tenant id to add')
+    .option('--shortcode <shortcode>', 'the organizations short code')
+    .option('--merchantname <merchantame>', 'the name given for the tenant')
+    .option('--tenantdescription <tenantdescription>', 'the tenant descriptions')
+    .option('--contact <contact>', 'Contact person to manage tenants')
+    .option('--email <email>', 'Email to contact')
+    .option('-j, --json', 'Formats the output in json')
+    .action(async function(options) {
+
+        var asJson = ( options.json ? options.json : false );
```
Comment (tobiaslohr) [inline cli.js:1941]: Do you use the `asJson` as all? If not we should remove it and from the options

**Review by tobiaslohr — CHANGES_REQUESTED** [review]
Thanks @hnestmann ! This is awesome. I've left some comments. In general (for all commands) we should consider offering an option for the user of the CLI to persist the short code via other means, e.g. local CLI config, env vars, .env or dw.json (or similar). 

Module `secrets` contains the general lookup logic already and we could use it to pull the `shortcode` to avoid having it being passed for every command.

File: cli.js (hunk 4 above)
Comment (hnestmann) [reply cli.js:1975]: not yet - I will add it though

File: cli.js (hunk 1 above)
Comment (hnestmann) [reply cli.js:1946]: I agree, here I have the flat list, but on client:add I reference a file. I think we should allow both approaches on both APIS.

File: cli.js (hunk 6 above)
Comment (hnestmann) [reply cli.js:1958]: this is now added

File: lib/slas.js [hunk 7]
```diff
@@ -0,0 +1,281 @@
+const fetch = require('node-fetch');
```
Comment (hnestmann) [inline lib/slas.js:1]: Add code comments

**Comment by hnestmann** [comment]
/home/circleci/project/cli.js
  1961:1  error  Line 1961 exceeds the maximum line length of 120  max-len

/home/circleci/project/lib/slas.js
   8:1  error  Line 8 exceeds the maximum line length of 120   max-len
  42:1  error  Line 42 exceeds the maximum line length of 120  max-len

**Comment by hnestmann** [comment]
> Thanks @hnestmann ! This is awesome. I've left some comments. In general (for all commands) we should consider offering an option for the user of the CLI to persist the short code via other means, e.g. local CLI config, env vars, .env or dw.json (or similar).
> 
> Module `secrets` contains the general lookup logic already and we could use it to pull the `shortcode` to avoid having it being passed for every command.

This has been added

**Review by tobiaslohr — APPROVED** [review]

**Pull request merged**